nimbusapp

```diff
@@ -140,30 +204,68 @@ if [[ -z ${PROJECT} ]]; then
+COMPOSE_FILE="${COMPOSE_DIR}/${IMAGE}.yml"
+COMPOSE_BACKUP="${COMPOSE_FILE}.bk"
 
-COMMAND="docker-app ${DOCKERAPP_ACTION}"
+if [[ -n "${DOCKERAPP_ACTION}" ]]; then
+   if curl -s --connect-timeout "${NIMBUS_DOCKERHUB_TIMEOUT}" "${NIMBUS_DOCKERHUB_URL}" > /dev/null 2>&1; then
+       DOCKERAPP_COMMAND="${DOCKERAPP_COMMAND} ${DOCKERAPP_ACTION} ${DOCKERAPP_ARGS} ${REPOSITORY}/${IMAGE}:${VERSION}"
 
-if [[ ! ${ARGS} == "" ]]; then
-    COMMAND="${COMMAND} ${ARGS}"
-fi
+       if [[ "$DOCKERAPP_ACTION" == "render" ]]; then
+          DOCKERAPP_COMMAND="${DOCKERAPP_COMMAND} > ${COMPOSE_FILE}"
+
+          # Backup previous successful attempts in case this try fails
+          # Ignore empty files (assume previous failure)
+          if [[ -s "${COMPOSE_FILE}" ]]; then
+            mv -v "${COMPOSE_FILE}" "${COMPOSE_BACKUP}"
+          else
+            rm -vf "${COMPOSE_FILE}"
+          fi
+       fi # action == render
+
+       mkdir -p "${COMPOSE_DIR}"
+
+       echo
+       echo "Running: ${DOCKERAPP_COMMAND}"
+
+       if ! eval "${DOCKERAPP_COMMAND}"; then
+          # Something went wrong, replace the compose backup
+          if [[ -f "${COMPOSE_BACKUP}" ]]; then
+            mv "${COMPOSE_BACKUP}" "${COMPOSE_FILE}"
+          fi
 
-COMMAND="${COMMAND} ${REPOSITORY}/${IMAGE}:${VERSION}"
+          echo "ERROR: Could not render" >&2
+          exit 1
+       fi # eval
 
-if [[ ${ISDOCKERAPPACTION} -eq 1 ]] ; then
-    if [[ -z ${VOLUME_MOUNT} ]]; then
-        COMMAND="${COMMAND} | docker-compose -p ${PROJECT} -f - ${ACTION} $@"
-    else
-       # TODO: think of a better way to provide this capability
-       COMMAND="${COMMAND} | sed -e "\$a\    - ${HOME}/.m2:/home/demo/.m2" -e "\$a\    - ${HOME}/${IDEA}:/home/demo/IdeaProjects" | docker-compose -p ${PROJECT} -f - ${ACTION} $@"
-    fi
+       if [[ -n "${VOLUME_MOUNT_IDEA}" ]]; then
+         echo "Mounting IDEA: ${VOLUME_MOUNT_IDEA}"
+         sed -i "/volumes:/a ${VOLUME_MOUNT_IDEA}" "${COMPOSE_FILE}"
+       fi
+
+       if [[ -n "${VOLUME_MOUNT_M2}" ]]; then
+         echo "Mounting M2: ${VOLUME_MOUNT_M2}"
+         sed -i "/volumes:/a ${VOLUME_MOUNT_M2}" "${COMPOSE_FILE}"
+       fi
+
+
+       if [[ "$DOCKERAPP_ACTION" == "render" && -z "${COMPOSE_ACTION}" ]]; then
+          cat "${COMPOSE_FILE}"
+       fi
+   else # curl check
+       if [[ ! -f "${COMPOSE_FILE}" ]]; then
+          echo "ERROR: No connection to Docker Hub: ${NIMBUS_DOCKERHUB_URL} (${NIMBUS_DOCKERHUB_TIMEOUT}s)" >&2
+          exit 1
+       else
+          echo "No connection to Docker Hub (${NIMBUS_DOCKERHUB_URL}), using cached file: ${COMPOSE_FILE}" >&2
+       fi
+   fi # curl check
 fi
 
-echo -e "Running ${COMMAND}"
-eval ${COMMAND}
\ No newline at end of file
+if [[ -n "${COMPOSE_ACTION}" ]]; then
+  COMPOSE_COMMAND="${COMPOSE_COMMAND} -p ${PROJECT} -f ${COMPOSE_FILE} ${COMPOSE_ACTION} ${COMPOSE_ARGS}"
+  echo "Running: ${COMPOSE_COMMAND}"
```

Comment:
We might need to add verbose messaging as it's starting to look spammy with the extra message talking about the caching. Thoughts?

Reply:
Agreed, I'll see about adding something, in most cases I think all the user will want is the output of the docker-compose

Reply:
Agreed, I think for most cases, users will only really need the output of `docker-compose`, I'll see about adding some verbosity settings